tests/test_api.py

```diff
@@ -0,0 +1,169 @@
+        self.progress(f"Uploading file {file.name} as {file.uuid} ")
+        response = self.assertPutResponse(
+            f"/v1/files/{file.uuid}",
+            requests.codes.created,
+            json_request_body=dict(
+                bundle_uuid=file.bundle.uuid,
+                creator_uid=0,
+                source_url=file.url
+            )
+        )
+        response_data = json.loads(response[1])
+        self.assertIs(type(response_data), dict)
+        self.assertIn('version', response_data)
+        self.progress(f"-> {response_data['version']}\n")
+        return response_data['version']
+
+    def create_bundle(self, bundle):
+        self.progress(f"Creating bundle {bundle.uuid} ")
+        response = self.assertPutResponse(
+            str(UrlBuilder().set(path='/v1/bundles/' + bundle.uuid).add_query('replica', 'aws')),
+            requests.codes.created,
+            json_request_body=self.put_bundle_payload(bundle)
+        )
+        response_data = json.loads(response[1])
+        self.assertIs(type(response_data), dict)
+        self.assertIn('version', response_data)
+        bundle.version = response_data['version']
+        self.progress(f"-> {bundle.version}\n")
+
+    @staticmethod
+    def put_bundle_payload(bundle):
+        payload = {
+            'uuid': bundle.uuid,
+            'creator_uid': 0,
+            'version': bundle.version,
+            'files': []
+        }
+        for file in bundle.files:
+            payload['files'].append(
+                {
+                    'indexed': True,
+                    'name': file.name,
+                    'uuid': file.uuid,
+                    'version': file.version
+                }
+            )
+        return payload
+
+    def get_bundle_and_check_files(self, bundle):
+        self.progress(f"Checking bundle {bundle.uuid}\n")
+        response = self.assertGetResponse(
+            str(UrlBuilder().set(path='/v1/bundles/' + bundle.uuid).add_query('replica', 'aws')),
+            requests.codes.ok
+        )
+        response_data = json.loads(response[1])
+        self.check_bundle_contains_same_files(bundle, response_data['bundle']['files'])
+        self.check_files_are_associated_with_bundle(bundle)
+
+    def check_bundle_contains_same_files(self, bundle, file_metadata):
+        self.assertEqual(len(bundle.files), len(file_metadata))
+        for file in bundle.files:
+            try:
+                filedata = next(data for data in file_metadata if data['uuid'] == file.uuid)
+            except StopIteration:
+                self.fail(f"File {file.uuid} is missing from bundle")
+            self.assertEqual(filedata['uuid'], file.uuid)
+            self.assertEqual(filedata['name'], file.name)
+            self.assertEqual(filedata['version'], file.version)
+
+    def check_files_are_associated_with_bundle(self, bundle):
+        for file in bundle.files:
+            response = self.assertGetResponse(
```

Comment:
No files are downloaded.  We're checking for a 302 response.  Are you saying that using GET causes extra processing on the back-end that we want to avoid?